diesel/src/pg/connection/stmt/mod.rs

```diff
@@ -68,31 +88,44 @@ impl Statement {
             PrepareForCache::No => None,
         };
         let name = query_name.as_deref();
-        let name = CString::new(name.unwrap_or(""))?;
-        let sql = CString::new(sql)?;
+        let name_cstr = CString::new(name.unwrap_or(""))?;
+        let sql_cstr = CString::new(sql)?;
         let param_types_vec = param_types
             .iter()
             .map(|x| x.oid())
             .collect::<Result<Vec<_>, _>>()
             .map_err(|e| crate::result::Error::SerializationError(Box::new(e)))?;
 
+        // For unnamed statements, we'll return a Statement object without actually preparing
+        if query_name.is_none() {
+            return Ok(Statement {
+                name: name_cstr,
+                param_formats: vec![1; param_types.len()],
+                sql: Some(sql_cstr),
+                param_types: Some(param_types_vec),
+            });
+        }
+
+        // For named/cached statements, prepare as usual
         let internal_result = unsafe {
             let param_count: libc::c_int = param_types
                 .len()
                 .try_into()
                 .map_err(|e| crate::result::Error::SerializationError(Box::new(e)))?;
             raw_connection.prepare(
-                name.as_ptr(),
-                sql.as_ptr(),
+                name_cstr.as_ptr(),
+                sql_cstr.as_ptr(),
                 param_count,
                 param_types_to_ptr(Some(&param_types_vec)),
             )
         };
         PgResult::new(internal_result?, raw_connection)?;
 
         Ok(Statement {
-            name,
+            name: name_cstr,
             param_formats: vec![1; param_types.len()],
+            sql: None,
+            param_types: None,
```

Comment:
Could we refactor this code a bit to better use the following pattern:

```
if let Some(name) = name {
   // do named statement stuff here
} else {
   // do unnamed stuff here
}
```

That would at least initializing the name with nothing.

Also it might be worth to introduce an enum 

```rust
enum StatementKind {
    Unnamed {sql: CString, param_types: Vec<_>},
    Named {name: CString}
}
```

to better group data that belong together. This enum could be used as a second field inside of `Statement` beside the `param_formats` field which is always the same for all statements.